### PR TITLE
[FEATURE] Add dynamic return type extension for UserAspect::get()

### PR DESCRIPTION
**Pull request by alexanderschnitzler**
Hi,
this is a dynamic return type extension for `UserAspect::get()`. There are dedicated getters but since `get()` exists and is in use, I'd like to have phpstan type support here.

Thanks for having a look.
Cheers. :)

### CODE REVIEW

File: src/Type/UserAspectGetDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+		$argumentType = $scope->getType($firstArgument->value);
+
+		if ($argumentType instanceof ConstantStringType) {
+			return match ($argumentType->getValue()) {
+				'id' => IntegerRangeType::createAllGreaterThanOrEqualTo(0),
+				'username' => new StringType(),
+				'isLoggedIn', 'isAdmin' => new BooleanType(),
+				'groupIds' => new ArrayType(new IntegerType(), IntegerRangeType::fromInterval(-2, null)),
+				'groupNames' => new ArrayType(new IntegerType(), new StringType()),
```
Comment (oliverklee) [inline src/Type/UserAspectGetDynamicReturnTypeExtension.php:51]: I think we can solve this with a stub (or better annotations in the Core) instead using https://phpstan.org/blog/phpstan-1-6-0-with-conditional-return-types, which would require far less code.

Comment (alexanderschnitzler) [reply src/Type/UserAspectGetDynamicReturnTypeExtension.php:51]: If you can tell me how this works, I am glad to implement it differently. AFAIK, you can only add one condition to implement a conditional return type like `@phpstan-return $raw is true ? array : QueryResultInterface`. Enlighten me please. :pray:

**Comment by sabbelasichon** [comment]
@alexanderschnitzler Thanks a lot. This is very useful indeed. We are still supporting PHP 7.4. So no match yet. 
@oliverklee IMHO: I don´t care about the amount of code, but prefer readibility and maintainibility. The conditional return types should be for simple use cases. I think the code is good as it is.

File: src/Type/UserAspectGetDynamicReturnTypeExtension.php (hunk 1 above)
Comment (alexanderschnitzler) [reply src/Type/UserAspectGetDynamicReturnTypeExtension.php:51]: That's a bit chaotic with 5 possible conditions, don't you think?

**Comment by alexanderschnitzler** [comment]
@sabbelasichon 

> @alexanderschnitzler Thanks a lot. This is very useful indeed. We are still supporting PHP 7.4. So no match yet.

Sure thing, will push a change as soon as the the discussion about other solutions is resolved.

File: src/Type/UserAspectGetDynamicReturnTypeExtension.php (hunk 1 above)
Comment (sabbelasichon) [reply src/Type/UserAspectGetDynamicReturnTypeExtension.php:51]: That´s what i think.

**Review by sabbelasichon — APPROVED** [review]

**Pull request merged**